src/NumericalAlgorithms/Spectral/Spectral.cpp

```diff
@@ -369,50 +369,72 @@ Matrix interpolation_matrix(const size_t num_points,
-         "since for Gauss-Lobatto you can just copy values off the volume.");
-  static const auto cache = make_static_cache<CacheRange<
-      1, Spectral::maximum_number_of_points<Spectral::Basis::Legendre>>>(
-      [](const size_t num_points) noexcept {
-        const Mesh<1> local_mesh{num_points, Spectral::Basis::Legendre,
-                                 Spectral::Quadrature::Gauss};
-        return std::pair<Matrix, Matrix>{interpolation_matrix(local_mesh, -1.0),
-                                         interpolation_matrix(local_mesh, 1.0)};
+    const size_t num_points) noexcept {
+  static_assert(BasisType == Spectral::Basis::Legendre);
+  static_assert(QuadratureType == Spectral::Quadrature::Gauss);
+  static const auto cache = make_static_cache<
+      CacheRange<Spectral::minimum_number_of_points<BasisType, QuadratureType>,
+                 Spectral::maximum_number_of_points<BasisType>>>(
+      [](const size_t local_num_points) noexcept {
+        return std::pair<Matrix, Matrix>{
+            interpolation_matrix<BasisType, QuadratureType>(local_num_points,
+                                                            -1.0),
+            interpolation_matrix<BasisType, QuadratureType>(local_num_points,
+                                                            1.0)};
       });
-  return cache(mesh.number_of_grid_points());
+  return cache(num_points);
 }
 
-const std::pair<DataVector, DataVector>& boundary_lifting_term(
+const std::pair<Matrix, Matrix>& boundary_interpolation_matrix(
     const Mesh<1>& mesh) noexcept {
   ASSERT(mesh.basis(0) == Spectral::Basis::Legendre,
          "We only support DG with a Legendre basis.");
   ASSERT(mesh.quadrature(0) == Spectral::Quadrature::Gauss,
          "We only compute the boundary interpolation for Gauss quadrature "
          "since for Gauss-Lobatto you can just copy values off the volume.");
-  static const auto cache = make_static_cache<CacheRange<
-      1, Spectral::maximum_number_of_points<Spectral::Basis::Legendre>>>(
-      [](const size_t num_points) noexcept {
-        const Mesh<1> local_mesh{num_points, Spectral::Basis::Legendre,
-                                 Spectral::Quadrature::Gauss};
-        const auto& matrices = boundary_interpolation_matrix(local_mesh);
-        std::pair<DataVector, DataVector> result{
-            DataVector{local_mesh.number_of_grid_points()},
-            DataVector{local_mesh.number_of_grid_points()}};
-        for (size_t i = 0; i < local_mesh.number_of_grid_points(); ++i) {
+  return boundary_interpolation_matrix<Spectral::Basis::Legendre,
+                                       Spectral::Quadrature::Gauss>(
+      mesh.extents(0));
+}
+
+template <Basis BasisType, Quadrature QuadratureType>
+const std::pair<DataVector, DataVector>& boundary_lifting_term(
+    const size_t num_points) noexcept {
+  static_assert(BasisType == Spectral::Basis::Legendre);
+  static_assert(QuadratureType == Spectral::Quadrature::Gauss);
```

Comment:
Could you move the error messages from the `ASSERT`s below to these `static_assert`s when you replace the function below with `SPECTRAL_QUANTITY_FOR_MESH`

Reply:
I can copy the quadrature one, the basis one seems rather straight forward, just a restatement in words of the requirement in the static assert